Tests/TSCBasicTests/ProcessTests.swift

```diff
@@ -87,11 +88,46 @@ class ProcessTests: XCTestCase {
                 """)
 
             try withCustomEnv(["PATH": tmpdir.pathString]) {
-                XCTAssertEqual(Process.findExecutable("nonExecutableProgram"), nil)
+                XCTAssertNil(Process.findExecutable("nonExecutableProgram"))
             }
         }
+      #else
+        try testWithTemporaryDirectory { tmpdir in
+            // Test System32 without .exe suffix.
+            XCTAssertNotNil(Process.findExecutable("cmd"))
+            
+            // Test Windows with .exe suffix.
+            XCTAssertNotNil(Process.findExecutable("explorer.exe"))
+
+            // Test non-existant programs.
+            XCTAssertNil(Process.findExecutable("nonExistantProgram"))
+            XCTAssertNil(Process.findExecutable(""))
+
+            // Create a bat file to test.
+            let tempExecutable = tmpdir.appending(component: "program.bat")
+            try localFileSystem.writeFileContents(tempExecutable, bytes: """
+                @echo off
+                exit
+
+                """)
+
+            // Create a non-executable file to test.
+            let tempNonExecutable = tmpdir.appending(component: "program.bc")
+            try localFileSystem.writeFileContents(tempNonExecutable, bytes: """
+                @echo off
+                exit
+
+                """)
+
+            try withCustomEnv(["PATH": tmpdir.pathString]) {
+                XCTAssertNotNil(Process.findExecutable("program.bat"))
+                XCTAssertNil(Process.findExecutable("program.bc"))
+            }
+        }
+      #endif
     }
 
+  #if !os(Windows) // Foundation treats all readable files as executable on Windows.
```

Comment:
Going to be fixed in apple/swift-corelibs-foundation#2956

Reply:
This can now be dropped.

Reply:
Dropped √